src/web/mjs/connectors/MangaParkEN.mjs

```diff
@@ -21,71 +21,80 @@ export default class MangaParkEN extends Connector {
 
     async _getMangas() {
         let mangaList = [];
-        const uri = new URL('/ajax.reader.home.release', this.url);
+        const uri = new URL('/browse?sort=name', this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'nav.d-md-block ul.pagination li:nth-last-child(2) a');
+        const pageCount = parseInt(data[0].text.trim());
+        for(let page = 1; page <= pageCount; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const uri = new URL('/browse?sort=name&page=' + page, this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, '#subject-list div.item');
+        return data.map( element => {
+            const a = element.querySelector('a.fw-bold');
+            this.cfMailDecrypt(a);
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(a, request.url),
+                title: a.text.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        let chapterList = [];
+        const mangaId = manga.id.match(/\/(\d+)\/?/)[1];
+        const uri = new URL('/ajax.reader.subject.episodes.by.latest', this.url);
         for (let page = '', run = true; run;) {
             const request = new Request(uri, {
                 method: 'POST',
-                body: JSON.stringify({ prevPos: page }),
-                headers: { 'content-type': 'application/json' }
+                body: JSON.stringify({
+                    iid: mangaId,
+                    prevPos: page
+                }),
+                headers: {
+                    'Content-Type': 'application/json;charset=UTF-8'
+                }
             });
             const data = await this.fetchJSON(request);
-            const mangas = [...this.createDOM(data.html).querySelectorAll('div.item > div > a.fw-bold')].map(element => {
+            const chapters = [...this.createDOM(data.html).querySelectorAll('div.episode-item > div > a.chapt')].map(element => {
+                let lang = this.getRootRelativeOrAbsoluteLink(element, this.url).match(/c[\d.]+-(\w+)-i\d+/)[1];
```

Comment:
May use `const link = `this.getRootRelativeOrAbsoluteLink(element, this.url);` as intermediate variable, to prevent duplicated code

Reply:
Fixed